nixos/modules/config/theming.nix

```diff
@@ -0,0 +1,332 @@
+    optional (opt != null && opt.package != null) opt.package;
+in
+{
+  options = {
+    theming = {
+      gtk = {
+        enable = mkEnableOption "Gtk3 theming configuration";
+
+        font = mkOption {
+          type = types.nullOr fontType;
+          default = null;
+          description = ''
+            The font to use in GTK+ 3 applications.
+          '';
+        };
+
+        iconTheme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The icon theme to use.";
+        };
+
+        theme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The GTK+3 theme to use.";
+        };
+      };
+
+      qt = {
+        enable = mkEnableOption "Qt5 theming configuration";
+
+        font = mkOption {
+          type = types.nullOr fontType;
+          default = null;
+          description = ''
+            The font to use in Qt 5 applications.
+          '';
+        };
+
+        iconTheme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The icon theme to use.";
+        };
+
+        theme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The Qt5 theme to use.";
+        };
+
+        scheme = mkOption {
+          type = types.nullOr themeType;
+          default = null;
+          description = "The Qt5 color scheme to use.";
+        };
+      };
+
+      theme = mkOption {
+        type = types.nullOr (types.enum [
+          "adwaita"
+          "breeze"
+          "breezedark"
+        ]);
+        default = null;
+        description = ''
+          Predefined themes that are defaults in Plasma and Gnome environments.
+        '';
+      };
+
+    };
+  };
+
+  config = mkMerge [
+    
+    (mkIf gcfg.enable
+      (let
+        settings =
+          optionalAttrs (gcfg.font != null)
+            { gtk-font-name = gcfg.font.name; }
+          //
+          optionalAttrs (gcfg.theme != null)
+            { gtk-theme-name = gcfg.theme.name; }
+          //
+          optionalAttrs (gcfg.iconTheme != null)
+            { gtk-icon-theme-name = gcfg.iconTheme.name; };
+
+        keyfile =
+          optionalAttrs (gcfg.font != null)
+            { font-name = gcfg.font.name; }
+          //
+          optionalAttrs (gcfg.theme != null)
+            { gtk-theme = gcfg.theme.name; }  
+          // 
+          optionalAttrs (gcfg.iconTheme != null)
+            { icon-theme = gcfg.iconTheme.name; };
+
+      in {
+        environment.systemPackages =
+            optionalPackage gcfg.font
+            ++ optionalPackage gcfg.theme
+            ++ optionalPackage gcfg.iconTheme;
+
+        environment.etc."xdg/gtk-2.0/gtkrc".text =
+          concatStringsSep "\n" (
+            mapAttrsToList toGtk2Ini settings
+          );
+
+        environment.etc."xdg/gtk-3.0/settings.ini".text =
```

Comment:
Is this really working? I have to symlink it to $XDG_CONFIG_HOME/gtk-3.0 to get applications to apply these settings, at least since the last time I tried.

Reply:
@rnbmjoj works here. I Will try in VM for sure

Reply:
Can we get tests?

Reply:
Sorry, it does work: I was confusing `gtk.css` with `settings.ini`. That is not loaded from /etc/xdg/gtk-3.0, unfortunately.